notion-server/src/ethereum/mod.rs

```diff
@@ -403,21 +437,42 @@ impl ShardChain {
     }
 }
 
+#[derive(Default, PartialEq, Copy, Clone, Debug)]
+struct Bytes32 {
+    pub bytes: [u8; 32],
```

Comment:
Should probably be written as `Bytes32([u8; 32])` since naming this field doesn't really add value.

Reply:
I have restructured the 32-bytes conversion more generally to now be a `Vec<u8>` trait.